programs/fundamentals/math.sc

```diff
@@ -42,6 +42,35 @@ distance(vec1, vec2) -> sqrt(reduce(vec1 - vec2, _a + _*_, 0));
 
 dot(vec1, vec2) -> reduce(vec1 * vec2, _a + _, 0);
 
+norm(vec) -> sqrt(reduce(vec, _a + _*_, 0));
+
+normalize(vec) -> vec / sqrt(reduce(vec, _a + _*_, 0));
+
+cross(v, w) -> [v:1*w:2 - v:2*w:1, v:2*w:0 - v:0*w:2, v:0*w:1 - v:1*w:0];
```

Comment:
is there a more generalized algorithm for the cross product of vectors? cos in that case I would prefer to use that, but this is fine till then.

Reply:
what do you mean? Cross product is only an operation that exists in R³ between vectors. You can think of it in terms of the determinant of a matrix where the first row is the versors of the three directions and the second and third are the multiplied vetors, but that wouldn't be any faster than this.

Reply:
ah k

Reply:
Wait, apparently you can make cross product for vectors of length>3, but it's a pain, and I also just realised ur only doing vectors of length 3 here so nvm